services/QuillLMS/app/controllers/grades_controller.rb

```diff
@@ -46,7 +46,7 @@ def tooltip
       percentage: activity_session.percentage,
       description: activity_session.activity.description,
       due_date: unit_activity.due_date,
-      completed_at: activity_session.completed_at.to_i + current_user.utc_offset.seconds,
+      completed_at: Time.at(activity_session.completed_at.to_i + current_user.utc_offset.seconds),
```

Comment:
Probably not going to happen, but you could add the save navigation operator

`activity_session&.completed_at ...`

Reply:
I don't think that's necessary here since this method is only called on a `.map` of `ActivitySession` records from the database, and I don't want to add too much unnecessary branching complexity, but thank you for flagging.